app/controllers/post_controller.rb

```diff
@@ -41,14 +42,18 @@ def submit
   end
 
   private
-    def validate_submission_parameters(title, author, markdown_text)
+    def validate_submission_parameters(title, author, markdown_text, hero)
       validation_message = nil
       if title.empty?
         validation_message = 'A post cannot be submited with a blank title.'
       elsif author.empty?
         validation_message = 'A post cannot be submited without an author.'
       elsif markdown_text.empty?
-        validation_message = 'A post cannot be submited with no markdown content.'
+        validation_message = 'A post cannot be submited with no markdown content.' 
+      elsif !hero.empty? && !(hero =~ URI::regexp)
```

Comment:
Please add a test case here for the submit method for an invalid URL. 

The test block should be something like `test 'post/submit should redirect back to the edit screen with an error message if a post was submited with a hero that's an invalid URL.'`

If you need an example of what you're test should look like. See the test block `test 'post/submit should redirect back to the edit screen with an error message if a post was submited without markdown'` in post_controller_test.rb

Reply:
Also, please add a test case here for when the hero is a valid URL. I think all of the tests currently use empty strings as a hero.